pkg/app/appserver/proc.go

```diff
@@ -62,12 +64,21 @@ func (p *Proc) Start() error {
 
 // Stop stops the application.
 func (p *Proc) Stop() error {
-	return p.cmd.Process.Signal(os.Interrupt)
+	err := p.cmd.Process.Signal(os.Interrupt)
+	if err != nil {
+		return err
+	}
+
+	<-p.doneC
+
+	return nil
 }
 
 // Wait waits for the application cmd to exit.
 func (p *Proc) Wait() error {
```

Comment:
`Wait()` should only be called once, and `Stop()` will only return if `Wait()` is called before. If this is by design, we should include it in the comments.

If not, an alternative would be to protect `p.cmd.Wait()` with `sync.Mutex`. And use it in both `Stop()` and `Wait()` (instead of having `doneC chan struct{}`).

Let me know if I'm mistaken.

Reply:
@evanlinjin well spotted! I'll fix this stuff